packages/patternfly-4/react-core/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import styles from '@patternfly/patternfly/components/Button/button.css';
+import { css, getModifier } from '@patternfly/react-styles';
+import { OneOf } from '../../helpers/typeUtils';
+
+export const ButtonVariant = {
+  primary: 'primary',
+  secondary: 'secondary',
+  tertiary: 'tertiary',
+  danger: 'danger',
+  link: 'link',
+  plain: 'plain'
+};
+
+export const ButtonType = {
+  button: 'button',
+  submit: 'submit'
+};
+
+export interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
+  /** content rendered inside the button */
+  children?: React.ReactNode;
+  /** additional classes added to the button */
+  className?: string; 
+  /** Sets the base component to render. defaults to button */
+  component?: React.ReactType | string;  
```

Comment:
React.ReactType has been deprecated.  We should be using React.ElementType.   React.ElementType includes strings, so you can get rid of the `| string`